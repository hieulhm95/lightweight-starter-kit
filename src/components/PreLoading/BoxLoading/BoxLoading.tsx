import React from 'react';
import ContentLoader from 'react-content-loader';
import theme from '@clad-ui/theme';

const { colors } = theme;

/**
 * @param {object} props
 * @param {object} [props.style] extra style
 * @param {string} [props.id] Use the same value of prop key, set this to solve inconsistency on the SSR and client
 */
const BoxLoading = ({ style = {}, id = '' }: { style?: object; id?: string }) => (
  <ContentLoader
    style={style}
    height={400}
    width={400}
    speed={2}
    uniqueKey={id}
    foregroundColor={colors.background}
    backgroundColor={colors.neutral}
  >
    <rect x="75" y="233" rx="4" ry="4" width="100" height="13" />
    <rect x="75" y="260" rx="4" ry="4" width="50" height="8" />
    <rect x="0" y="210" rx="5" ry="5" width="400" height="10" />
    <rect x="0" y="0" rx="5" ry="5" width="400" height="200" />
  </ContentLoader>
);

export default BoxLoading;
