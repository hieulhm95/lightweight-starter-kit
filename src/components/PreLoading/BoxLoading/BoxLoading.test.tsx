import React from 'react';
import { render /* screen */ } from '@testing-library/react';
import BoxLoading from './BoxLoading';

describe('BoxLoading component', () => {
  it('should match snapshot', () => {
    const wrapper = render(<BoxLoading id="abc" />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('should render with inline styles', () => {
    const wrapper = render(<BoxLoading id="abc" style={{ width: '100%', height: '300px' }} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
