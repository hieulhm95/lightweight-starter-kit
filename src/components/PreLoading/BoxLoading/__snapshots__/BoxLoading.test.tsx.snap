// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxLoading component should match snapshot 1`] = `
<DocumentFragment>
  <svg
    aria-labelledby="abc-aria"
    height="400"
    role="img"
    width="400"
  >
    <title
      id="abc-aria"
    >
      Loading...
    </title>
    <rect
      clip-path="url(#abc-diff)"
      height="100%"
      role="presentation"
      style="fill: url(#abc-animated-diff);"
      width="100%"
      x="0"
      y="0"
    />
    <defs
      role="presentation"
    >
      <clippath
        id="abc-diff"
      >
        <rect
          height="13"
          rx="4"
          ry="4"
          width="100"
          x="75"
          y="233"
        />
        <rect
          height="8"
          rx="4"
          ry="4"
          width="50"
          x="75"
          y="260"
        />
        <rect
          height="10"
          rx="5"
          ry="5"
          width="400"
          x="0"
          y="210"
        />
        <rect
          height="200"
          rx="5"
          ry="5"
          width="400"
          x="0"
          y="0"
        />
      </clippath>
      <lineargradient
        id="abc-animated-diff"
      >
        <stop
          offset="0%"
          stop-color="#C0C0C0"
          stop-opacity="1"
        >
          <animate
            attributeName="offset"
            dur="2s"
            keyTimes="0; 0.25; 1"
            repeatCount="indefinite"
            values="-2; -2; 1"
          />
        </stop>
        <stop
          offset="50%"
          stop-color="#F4F4F4"
          stop-opacity="1"
        >
          <animate
            attributeName="offset"
            dur="2s"
            keyTimes="0; 0.25; 1"
            repeatCount="indefinite"
            values="-1; -1; 2"
          />
        </stop>
        <stop
          offset="100%"
          stop-color="#C0C0C0"
          stop-opacity="1"
        >
          <animate
            attributeName="offset"
            dur="2s"
            keyTimes="0; 0.25; 1"
            repeatCount="indefinite"
            values="0; 0; 3"
          />
        </stop>
      </lineargradient>
    </defs>
  </svg>
</DocumentFragment>
`;

exports[`BoxLoading component should render with inline styles 1`] = `
<DocumentFragment>
  <svg
    aria-labelledby="abc-aria"
    height="400"
    role="img"
    style="width: 100%; height: 300px;"
    width="400"
  >
    <title
      id="abc-aria"
    >
      Loading...
    </title>
    <rect
      clip-path="url(#abc-diff)"
      height="100%"
      role="presentation"
      style="fill: url(#abc-animated-diff);"
      width="100%"
      x="0"
      y="0"
    />
    <defs
      role="presentation"
    >
      <clippath
        id="abc-diff"
      >
        <rect
          height="13"
          rx="4"
          ry="4"
          width="100"
          x="75"
          y="233"
        />
        <rect
          height="8"
          rx="4"
          ry="4"
          width="50"
          x="75"
          y="260"
        />
        <rect
          height="10"
          rx="5"
          ry="5"
          width="400"
          x="0"
          y="210"
        />
        <rect
          height="200"
          rx="5"
          ry="5"
          width="400"
          x="0"
          y="0"
        />
      </clippath>
      <lineargradient
        id="abc-animated-diff"
      >
        <stop
          offset="0%"
          stop-color="#C0C0C0"
          stop-opacity="1"
        >
          <animate
            attributeName="offset"
            dur="2s"
            keyTimes="0; 0.25; 1"
            repeatCount="indefinite"
            values="-2; -2; 1"
          />
        </stop>
        <stop
          offset="50%"
          stop-color="#F4F4F4"
          stop-opacity="1"
        >
          <animate
            attributeName="offset"
            dur="2s"
            keyTimes="0; 0.25; 1"
            repeatCount="indefinite"
            values="-1; -1; 2"
          />
        </stop>
        <stop
          offset="100%"
          stop-color="#C0C0C0"
          stop-opacity="1"
        >
          <animate
            attributeName="offset"
            dur="2s"
            keyTimes="0; 0.25; 1"
            repeatCount="indefinite"
            values="0; 0; 3"
          />
        </stop>
      </lineargradient>
    </defs>
  </svg>
</DocumentFragment>
`;
